__tests__/app.test.js

```diff
@@ -72,79 +72,224 @@ describe("200: /api", () => {
+            expect(articles).toBeSorted({
+              key: "topic",
+              descending: true,
+            });
+          });
+      });
+      test("should allow user sort by article_id", () => {
+        return request(app)
+          .get("/api/articles?sort_by=article_id")
+          .expect(200)
+          .then(({ body }) => {
+            const { articles } = body;
+            articles;
+            expect(articles).toBeSorted({
+              key: "article_id",
+              descending: true,
+            });
+          });
+      });
+      test("should allow user sort by votes", () => {
+        return request(app)
+          .get("/api/articles?sort_by=votes")
+          .expect(200)
+          .then(({ body }) => {
+            const { articles } = body;
+            expect(articles).toBeSorted({
+              key: "votes",
+              descending: true,
+            });
+          });
+      });
+      test("should allow user sort by article_img_url", () => {
+        return request(app)
+          .get("/api/articles?sort_by=article_img_url")
+          .expect(200)
+          .then(({ body }) => {
+            const { articles } = body;
+            expect(articles).toBeSorted({
+              key: "article_img_url",
+              descending: true,
+            });
+          });
       });
+      test("should allow user sort by comment_count", () => {
+        return request(app)
+          .get("/api/articles?sort_by=comment_count")
+          .expect(200)
+          .then(({ body }) => {
+            const { articles } = body;
+            expect(articles).toBeSorted({
+              key: "comment_count",
+              descending: true,
+            });
+          });
+      });
+      test("should allow user sort by author", () => {
+        return request(app)
+          .get("/api/articles?sort_by=author")
+          .expect(200)
+          .then(({ body }) => {
+            const { articles } = body;
+            expect(articles).toBeSorted({
+              key: "author",
+              descending: true,
+            });
+          });
+      });
+      test("should allow user sort by title", () => {
+        return request(app)
+          .get("/api/articles?sort_by=title")
+          .expect(200)
+          .then(({ body }) => {
+            const { articles } = body;
+            expect(articles).toBeSorted({
+              key: "title",
+              descending: true,
+            });
+          });
+      });
+    });
+    test('should get articles by topic "cats"', () => {
+      return request(app)
+        .get("/api/articles?topic=cats")
+        .expect(200)
+        .then(({ body }) => {
+          const { articles } = body;
+          articles.forEach((article) => {
+            expect(article.topic).toBe("cats");
+          });
+        });
+    });
+    test("400 should return error for invalid query", () => {
+      return request(app).get("/api/articles?sort_by=cats").expect(400);
+    });
```

Comment:
Can we just have a few more tests please? Think about what we'd expect for the following scenarios...

-  Invalid order query, e.g. ?order=bananas
- Valid topic query, but has no articles e.g. ?topic=paper. 
-  Non-existent topic query, e.g. ?topic=bananas.